sonar-batch/src/main/java/org/sonar/batch/phases/DecoratorsExecutor.java

```diff
@@ -43,21 +43,23 @@
 
 public class DecoratorsExecutor implements BatchComponent {
 
-  private DecoratorsSelector decoratorsSelector;
-  private SonarIndex index;
-  private EventBus eventBus;
-  private Project project;
-  private CoverageExclusions coverageFilter;
-  private MeasureCache measureCache;
-  private MetricFinder metricFinder;
+  private final DecoratorsSelector decoratorsSelector;
+  private final SonarIndex index;
+  private final EventBus eventBus;
+  private final Project project;
+  private final CoverageExclusions coverageFilter;
+  private final MeasureCache measureCache;
+  private final MetricFinder metricFinder;
   private final DuplicationCache duplicationCache;
+  private final AnalysisMode analysisMode;
 
   public DecoratorsExecutor(BatchExtensionDictionnary batchExtDictionnary,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/henryju/image-hosting/master/major.png)Method has 9 parameters, which is greater than 7 authorized.[![...](https://raw.githubusercontent.com/henryju/image-hosting/master/more.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)